Assets/00_Content/Scripts/Menu/PlayerSlotObject.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+using Player;
+
+namespace Menu {
+
+    public class PlayerSlotObject : MonoBehaviour
+    {
+
+        public int Id;
+
+        private bool isTaken;
+        public bool IsTaken => isTaken;
+
+        private PlayerComponent playerComponent;
+        public PlayerComponent PlayerComponent => playerComponent;
+
+        [SerializeField] TextMeshProUGUI joinText;
+        [SerializeField] TextMeshProUGUI readyText;
+
+        private Image background;
```

Comment:
The order of these fields is a bit confusing, perhaps a reordering would improve clarity